contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -49,11 +64,12 @@ select
   , CAST(null as integer) as history_table_id
   , CAST(0 as sys.bit) as is_remote_data_archive_enabled
   , CAST(0 as sys.bit) as is_external
-from pg_class t inner join sys.schemas sch on t.relnamespace = sch.schema_id
-where t.relpersistence in ('p', 'u', 't')
+from pg_class t 
+where t.relnamespace in (select schema_id from sys.schemas)
+and t.relpersistence in ('p', 'u', 't')
 and t.relkind = 'r'
-and not sys.is_table_type(t.oid)
-and has_schema_privilege(sch.schema_id, 'USAGE')
+and t.oid not in (select typrelid from sys.table_types_internal)
```

Comment:
I wonder if we could use with clause to avoid another internal object here .

Reply:
Do you mean to define sys.table_types_internal  a cte (with clause) locally in this view instead having a separate internal view?

In this case I think having sys.table_types_internal view defined separately is beneficial because it is used in more than one view. In this CR it is used in sys.tables and sys.types view.

It potentially can be also used in many places where the is_table_type function is currently used. 
Prior, to modifications in this PR calling the is_table_type function was very expensive in sys.types view as it was called for every row (that matched) and internally it is doing select/join.

For example simple 
select * from sys.types 
called the sys.is_table_type function 2680 times.

Again, all the context is in this quip doc:
https://quip-amazon.com/kSRrAAk6bxJR#ZeC9AAQzDNP